crypto/encode_decode/decoder_pkey.c

```diff
@@ -246,10 +280,30 @@ static void collect_decoder(OSSL_DECODER *decoder, void *arg)
     end_i = sk_OPENSSL_CSTRING_num(data->names);
     for (i = 0; i < end_i; i++) {
         const char *name = sk_OPENSSL_CSTRING_value(data->names, i);
+        OSSL_DECODER_INSTANCE *di = NULL;
+
+        if (OSSL_DECODER_is_a(decoder, name)
+            /*
+             * Either the caller didn't give a selection, or if they did,
+             * the decoder must tell us if it supports that selection to
+             * be accepted.  If the decoder doesn't have |does|, it's seen
```

Comment:
So why not name it something like you have described instead of 'does'

supports_selection? maybe
OR has_selection

Reply:
`supports_selection` would be my choice, then...  or maybe `does_selection`? :laughing:

Reply:
either is better than does..My brain is fried for the day. I will re look at this PR another day.

Reply:
`does_selection` it is!